include/replayio-macros.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2024 Record Replay Inc.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// API for interacting with the record/replay driver.
+// Some parts are still in v8.h and still need to be migrated.
+
+
+#ifndef INCLUDE_REPLAYIO_MACROS_H_
+#define INCLUDE_REPLAYIO_MACROS_H_
+
+// Use this to wrap Asserts on non-trivial data, to avoid the
+// overhead of reads when Asserts are disabled.
+#define REPLAY_ASSERT(format, ...) \
+  if (recordreplay::HasAsserts()) \
+    recordreplay::Assert(format, ##__VA_ARGS__); \
+  static_assert(true, "require semicolon")
```

Comment:
huh, have never seen this used before.  `do { .. } while (0)` was usually the strategy.